connman/plugins/clat.c

```diff
@@ -1713,15 +1885,18 @@ static void stop_running(struct clat_data *data)
 	clat_task_stop_dad(data);
 	clat_task_stop_online_check(data);
 
-	/* When stopping always disable double nat but backup tethering state */
-	tethering = data->tethering;
-	data->tethering = TETHERING_OFF;
-	setup_double_nat(data);
-	data->tethering = tethering;
+	/* When stopping disable double nat but backup tethering state */
+	if (data->tethering == TETHERING_ON) {
+		tethering = data->tethering;
```

Comment:
```
static void clat_tethering_changed(struct connman_technology *tech, bool on)
{
	struct clat_data *data = get_data();

	/* We just need to know if tethering is enabled or not */
	data->tethering = on ? TETHERING_ON : TETHERING_OFF;

	setup_double_nat(data);
}
``` 

This follows tethering status. And the value is backed up because of the double nat cleanup that requires the state to be off in order to disable everything. And the value is restored to not to be in inconsistent state.

This could be cleaner by calling merely `connman_nat_disable_double_nat_override(TAYGA_CLAT_DEVICE);` but...